catalog/core/catalog-core-commands/src/test/groovy/org/codice/ddf/commands/catalog/ExportCommandSpec.groovy

```diff
@@ -0,0 +1,283 @@
+import ddf.catalog.operation.impl.ResourceResponseImpl
+import ddf.catalog.resource.ResourceNotFoundException
+import ddf.catalog.resource.impl.ResourceImpl
+import ddf.catalog.transform.MetacardTransformer
+import groovy.xml.MarkupBuilder
+import org.apache.karaf.shell.api.console.Session
+import org.osgi.framework.BundleContext
+import org.osgi.framework.ServiceReference
+
+import javax.activation.MimeType
+import java.nio.file.Paths
+import java.util.zip.ZipFile
+
+class ExportCommandSpec extends spock.lang.Specification {
+
+    ExportCommand exportCommand
+
+    CatalogFramework catalogFramework
+
+    MetacardTransformer xmlTransformer
+
+    File tmpHomeDir
+
+    void setup() {
+        tmpHomeDir = File.createTempDir()
+        System.setProperty("ddf.home", tmpHomeDir.canonicalPath)
+
+        ServiceReference xmlTransformerReference = Mock(ServiceReference)
+
+        xmlTransformer = Mock(MetacardTransformer)
+
+        xmlTransformer.transform(_ as Metacard, _ as Map) >> { metacard, map ->
+            metacardToXmlTransformer(metacard, map)
+        }
+
+        BundleContext bundleContext = Mock(BundleContext) {
+            getServiceReferences(MetacardTransformer, '(id=xml)') >> [xmlTransformerReference]
+            getService(xmlTransformerReference) >> xmlTransformer
+        }
+
+        catalogFramework = Mock(CatalogFramework)
+        catalogFramework.query(_ as QueryRequest) >> { QueryRequest req ->
+            new QueryResponseImpl(req, [], 0)
+        }
+        catalogFramework.getLocalResource(_ as ResourceRequest) >> {
+            throw new ResourceNotFoundException('Could not find exception')
+        }
+
+        exportCommand = new ExportCommand(filterBuilder: new GeotoolsFilterBuilder(),
+                bundleContext: bundleContext, catalogFramework: catalogFramework)
+
+    }
+
+    void cleanup() {
+        assert tmpHomeDir?.deleteDir()
+    }
+
+    def "Test export no items"() {
+        setup:
+        exportCommand.with {
+            delete = false
+        }
+
+        when:
+        exportCommand.executeWithSubject()
+
+        then:
+        notThrown(Exception)
+    }
+
+    def "Test filename that already exists"() {
+        setup:
+        def file = Paths.get(System.getProperty('ddf.home'), 'filealreadyexists').toFile()
+        file.createNewFile()
+        exportCommand.with {
+            delete = false
+            output = file.canonicalPath
+        }
+
+        when:
+        exportCommand.executeWithSubject()
+
+        then:
+        thrown(IllegalStateException)
+    }
+
+    def "Test bad file name"() {
+        setup:
+        def file = Paths.get(System.getProperty('ddf.home'), 'badFilename.notazip')
+        exportCommand.with {
+            delete = false
+            output = file
+        }
+
+        when:
+        exportCommand.executeWithSubject()
+
+        then:
+        thrown(IllegalStateException)
+    }
+
+    def "Test abort command"() {
```

Comment:
❓ Do we really need to test this? Feels like we're testing the Karaf console more than the actual command. Thoughts?